File: aidl_client/app/src/main/java/com/example/aidl_client/CustomRect.java
package com.example.aidl_client;

import android.os.Parcel;
import android.os.Parcelable;

public final class CustomRect implements Parcelable {
    public int left;
    public int top;
    public int right;
    public int bottom;

    public static final Parcelable.Creator<CustomRect> CREATOR = new Parcelable.Creator<CustomRect>() {
        public CustomRect createFromParcel(Parcel in) {
            return new CustomRect(in);
        }

        public CustomRect[] newArray(int size) {
            return new CustomRect[size];
        }
    };

    public CustomRect() {
    }

    private CustomRect(Parcel in) {
        readFromParcel(in);
    }

    public void writeToParcel(Parcel out, int flags) {
        out.writeInt(left);
        out.writeInt(top);
        out.writeInt(right);
        out.writeInt(bottom);
    }

    public void readFromParcel(Parcel in) {
        left = in.readInt();
        top = in.readInt();
        right = in.readInt();
        bottom = in.readInt();
    }

    public int describeContents() {
        return 0;
    }
}

File: aidl_server/app/src/main/java/com/example/aidl_server/CustomRect.kt
package com.example.aidl_server

import android.os.Parcel
import android.os.Parcelable

class CustomRect() : Parcelable {
    var left: Int = 0
    var top: Int = 0
    var right: Int = 0
    var bottom: Int = 0

    companion object CREATOR : Parcelable.Creator<CustomRect> {
        override fun createFromParcel(parcel: Parcel): CustomRect {
            return CustomRect(parcel)
        }

        override fun newArray(size: Int): Array<CustomRect?> {
            return Array(size) { null }
        }
    }

    private constructor(inParcel: Parcel) : this() {
        readFromParcel(inParcel)
    }

    override fun writeToParcel(outParcel: Parcel, flags: Int) {
        outParcel.writeInt(left)
        outParcel.writeInt(top)
        outParcel.writeInt(right)
        outParcel.writeInt(bottom)
    }

    private fun readFromParcel(inParcel: Parcel) {
        left = inParcel.readInt()
        top = inParcel.readInt()
        right = inParcel.readInt()
        bottom = inParcel.readInt()
    }

    override fun describeContents(): Int {
        return 0
    }
}